src/Form/Mapper/FormMapper.php

```diff
@@ -41,10 +41,11 @@ public function reorder(array $keys);
     /**
      * @param class-string<FormTypeInterface>|null $type
      * @param array<string, mixed>                 $options
+     * @param array<string, mixed>                 $fieldDescriptionOptions
      *
      * @return static
      */
-    public function add(string $name, ?string $type = null, array $options = []);
+    public function add(string $name, ?string $type = null, array $options = [], array $fieldDescriptionOptions = []);
```

Comment:
I think we might have a problem here, once I added the 4th parameter on containerBlockService, phpstan complains because that parameter does not exist on the interface, and he is right.

but fieldDescriptionOptions have the followin type on AdminBundle: 

```
 * @psalm-type FieldDescriptionOptions array{
 *     accessor?: string|callable|PropertyPathInterface,
 *     associated_property?: string|callable|PropertyPathInterface,
 *     label?: string|false|null,
 *     link_parameters?: array<string, mixed>,
 *     role?: string|string[],
 *     sort_field_mapping?: array<string, mixed>,
 *     sort_parent_association_mappings?: array<array<string, mixed>>,
 *     sortable?: string|bool,
 *     template?: string,
 *     translation_domain?: string|false,
 *     type?: string,
 *     virtual_field?: bool
 * }&array<string, mixed>
 * @phpstan-type FieldDescriptionOptions = array<string, mixed>
```

We will have trouble with PHPStan or Psalm because of that?

Reply:
The fieldDescrpition interface is on AdminBundle, and I don't think it makes sense to move it...

Reply:
We have the Admin FormMapper, extending the Block FormMapper
https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Form/FormMapper.php#L36

FieldDescription seems to me something only related to the Admin FormMapper. Why do you need to add a 4th parameter here ?
Can't we check it's an AdminFormMapper on PageBundle ?

Reply:
I need to add because I am using it here: https://github.com/sonata-project/SonataBlockBundle/pull/1100/files/bde0b23549e42018ceebf8f3b064dbcfb6f607e4#diff-0dc8b125bb9655cf156cd39bc2c8e992145d4337a2c3d0cd3a195b0661322e73

Reply:
And I can't check for that on PageBundle, since the block service is placed here, and here we don't depend on AdminBundle.

TBH, if we don't want to solve it this way, another option is to deprecate this block (remove it on 5.0) and implement it directly on BlockBundle.

Reply:
> I need to add because I am using it here: https://github.com/sonata-project/SonataBlockBundle/pull/1100/files/bde0b23549e42018ceebf8f3b064dbcfb6f607e4#diff-0dc8b125bb9655cf156cd39bc2c8e992145d4337a2c3d0cd3a195b0661322e73

I see. I think the issue is kinda in the ContainerService then.
This is not really meant to be used by something else than an admin if we start to add `'edit' => 'inline'` etc, because these options are supported by SonataAdmin.

The idea of having these BlockService here should be to be able to use BlockBundle without AdminBundle.
If some blocks are too much related they should be in AdminBundle or a SpecificBundle.

SeoBundle have specific services https://github.com/sonata-project/SonataSeoBundle/tree/3.x/src/Block/Breadcrumb

So I start to wondering if the ContainerService shouldn't be move to PageBundle then.
If it doesn't work without adding `edit => inline`, that would mean that nobody use it atm.

Reply:
The only thing I didn't tried is if the normal collection type with the modal works at all, I will try that one first, before deprecating this.

Reply:
> IMO, if we don't want to solve it this way, another option is to deprecate this block (remove it on 5.0) and implement it directly on PageBundle.

Thats exactly what I have in mind

Reply:
Another thing to take into account is that I am not sure if CollectionType is really useful outside Sonata:

1. https://github.com/sonata-project/form-extensions/blob/1.x/src/Type/CollectionType.php Seems really incomplete. It can only render textTypes without SonataAdmin?
2. When you use SonataAdmin, that CollectionType is related to the admin of the class you are doing collection, which we don't have here really (it fails if you don't have an admin)

So probably the best option is to deprecate this one, since it can't be used... but I will do some testing first.

Reply:
> Another thing to take into account is that I am not sure if CollectionType is really useful outside SonataAdmin and the concept of Admin and entity classes:
> 
> 1. https://github.com/sonata-project/form-extensions/blob/1.x/src/Type/CollectionType.php Seems really incomplete. It can only render textTypes without SonataAdmin?
> 2. When you use SonataAdmin, that CollectionType is related to the admin of the class you are doing collection, which we don't have here really (it fails if you don't have an admin)
> 
> So probably the best option is to deprecate this one, since it can't be used... but I will do some testing first.

If we dig more, there is something wrong with the doc https://docs.sonata-project.org/projects/form-extensions/en/1.x/reference/form_types/ because all the example are given in a SonataAdmin context.

And some template are in SonataAdmin https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Resources/views/Form/form_admin_fields.html.twig#L430-L444

I dunno if all the form in form-extensions can be used correctly without SonataAdmin.

Reply:
Some of them should be fine, like the date picker, but this one I am not sure it makes sense 😅

Reply:
Reverted this once